contracts/MerkleProof.sol

```diff
@@ -68,4 +68,29 @@ library MerkleProof {
 
     return result[0] == 0x01;
   }
+
+  /**
+   * @dev Returns true if a `leaf` can be proved to be a part of a Merkle tree
+   * defined by `root`. For this, a `proof` must be provided, containing
+   * sibling hashes on the branch from the leaf to the root of the tree. Each
+   * pair of leaves and each pair of pre-images are assumed to be sorted.
+   */
+  function verify(bytes32[] memory proof, bytes32 root, bytes32 leaf) internal pure returns (bool) {
```

Comment:
I suggest we use https://github.com/OpenZeppelin/openzeppelin-contracts/blob/6bc1173c8e37ca7de2201a0230bb08e395074da1/contracts/utils/cryptography/MerkleProof.sol#L32C1-L32C1, but not implement one by ourselfves.

Reply:
after upgrade to 0.8.17, I use the original openzeppelin lib